### PR TITLE
736: Allow only sweeping vrp type functional test coverage

### PR DESCRIPTION
**Pull request by frlaura-jianu**
Issue: https://github.com/SecureApiGateway/SecureApiGateway/issues/736
Description: Added the functional test for "allow only sweeping vrp type".

### CODE REVIEW

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/api/v3_1_10/CreateDomesticVrp.kt [hunk 1]
```diff
@@ -266,6 +268,22 @@ class CreateDomesticVrp(val version: OBVersion, val tppResource: CreateTppCallba
         assertThat((exception.cause as FuelError).response.responseMessage).isEqualTo(UNAUTHORIZED)
     }
 
+    fun shouldCreateDomesticVrpConsent_throwsBadRequestHasDifferentVrpTypeOtherThanSweepingTest() {
```
Comment (jorgesanchezperez) [inline src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/api/v3_1_10/CreateDomesticVrp.kt:271]: Maybe the method name could be simplified.

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/junit/v3_1_10/CreateDomesticVrpTest.kt [hunk 2]
```diff
@@ -105,6 +105,17 @@ class CreateDomesticVrpTest(val tppResource: CreateTppCallback.TppResource) {
...
     }
 
+    @EnabledIfVersion(
+        type = "payments",
+        apiVersion = "v3.1.10",
+        operations = ["CreateDomesticVrpPayment", "CreateDomesticVRPConsent", "GetDomesticVRPConsent"],
+        apis = ["domestic-vrps", "domestic-vrp-consents"]
+    )
+    @Test
+    fun shouldCreateDomesticVrp_throwsBadRequestHasDifferentVrpTypeOtherThanSweeping_v3_1_10() {
```
Comment (jorgesanchezperez) [inline src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/junit/v3_1_10/CreateDomesticVrpTest.kt:115]: maybe the name could be shorted.

**Review by jorgesanchezperez — CHANGES_REQUESTED** [review]
Make the method name shorted.
Add the same test in the versions v3.1.8 and 3.1.9 please.

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/api/v3_1_10/CreateDomesticVrp.kt (hunk 1 above)
Comment (frlaura-jianu) [reply src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/api/v3_1_10/CreateDomesticVrp.kt]: Made the requested changes.

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/junit/v3_1_10/CreateDomesticVrpTest.kt (hunk 2 above)
Comment (frlaura-jianu) [reply src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/domestic/vrp/junit/v3_1_10/CreateDomesticVrpTest.kt]: Made the requested changes.

**Comment by frlaura-jianu** [comment]
> Make the method name shorted. Add the same test in the versions v3.1.8 and 3.1.9 please.

We only support v3.1.10 for the domestic vrps functional tests.

**Review by dbadham-fr — APPROVED** [review]

**Review by jorgesanchezperez — APPROVED** [review]

**Pull request merged**